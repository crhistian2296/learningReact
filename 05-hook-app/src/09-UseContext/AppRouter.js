import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import AboutScreen from './AboutScreen';
import HomeScreen from './HomeScreen';
import LogginScreen from './LogginScreen';

const AppRouter = () => (
  <>
    <Router>
      <Routes>
        <Route path='/about' element={<AboutScreen />} />
        <Route path='/loggin' element={<LogginScreen />} />
        <Route path='/' element={<HomeScreen />} />
      </Routes>
    </Router>
  </>
);
export default AppRouter;
