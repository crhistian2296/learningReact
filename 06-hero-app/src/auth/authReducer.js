// const user = {
//   name:'Fernando',
//   logged: true
// }

import { login, logout } from '../types/type';

export const authReducer = (state = {}, action) => {
  switch (action.type) {
    case login:
      return {
        ...action.payload,
        logged: true,
      };
    case logout:
      return {
        logged: false,
      };
    default:
      return state;
  }
};
