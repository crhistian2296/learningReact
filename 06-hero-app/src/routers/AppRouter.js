import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import LoginScreen from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoutes } from './PrivateRoutes';
import { PublicRoutes } from './PublicRoutes';

const AppRouter = () => {
  return (
    <Router>
      <div className='main'>
        <Routes>
          {/* <Route path='login' element={<LoginScreen />} /> */}
          <Route
            path='login'
            element={
              <PublicRoutes>
                <LoginScreen />
              </PublicRoutes>
            }
          />
          <Route
            path='*'
            element={
              <PrivateRoutes>
                <DashboardRoutes />
              </PrivateRoutes>
            }
          />
        </Routes>
      </div>
    </Router>
  );

  // *react router v6 convention
  // return (
  //   <Router>
  //     <div className='main'>
  //       <Routes>
  //         <Route path='login' element={<LoginScreen />} />
  //         <Route path='' element={<PrivateRoutes />}>
  //           <Route element={<DashboardRoutes />}>
  //             <Route path='hero/:heroId' element={<HeroScreen />} />
  //             <Route path='marvel' element={<MarvelScreen />} />
  //             <Route path='dc' element={<DCScreen />} />
  //             <Route path='search' element={<SearchScreen />} />

  //             {/* Default */}
  //             <Route path='/' element={<MarvelScreen />} />
  //           </Route>
  //         </Route>
  //       </Routes>
  //     </div>
  //   </Router>
  // );
};

export default AppRouter;
