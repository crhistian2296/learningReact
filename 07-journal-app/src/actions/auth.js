import { firebase, googleAuthProvider } from '../firebase/firebase-config';
import { types } from '../types/types';
import { uiFinishLoading, uiStartLoading } from './ui';

// export const startLoginEmailPassword = (email, password) => {
//   // *redux-thunk provee automaticamente ese dispatch para acciones asincronas
//   return (dispatch) => {
//     setTimeout(() => {
//       dispatch(login(123, 'crhistian'));
//     }, 3000);
//   };
// };

export const login = (uid, displayName) => ({
  type: types.login,
  payload: {
    uid,
    displayName,
  },
});
/**
 * Crea una cuenta en la app de autenticacion en firebase mediante correo
 * @param {string} email
 * @param {string} password
 * @param {string} name
 * @returns void
 */
export const startRegisterWithEmailAsync = (email, password, name) => {
  return (dispatch) => {
    dispatch(uiStartLoading());
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });
        console.log(user);
        dispatch(login(user.uid, user.displayName));
      })
      .catch((err) => console.log(err))
      .finally(() => dispatch(uiFinishLoading()));
  };
};
export const startLoginWithEmailAsync = (email, password) => {
  return (dispatch) => {
    dispatch(uiStartLoading());
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        console.log(user);
        dispatch(login(user.uid, user.displayName));
      })
      .catch((err) => console.log(err))
      .finally(() => dispatch(uiFinishLoading()));
  };
};

/**
 * Crean una cuenta en la app de autenticacion de firebase mediante boton de google
 * @returns void
 */
export const startGoogleLogin = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user }) => dispatch(login(user.uid, user.displayName)))
      .catch(console.log);
  };
};

export const startLogout = () => {
  return async (dispatch) => {
    await firebase.auth().signOut();

    await dispatch(logout());
  };
};

export const logout = () => ({ type: types.logout });
