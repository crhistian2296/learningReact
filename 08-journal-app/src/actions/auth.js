import { login } from '../types/types';

export const loginAction = (uid, displayName) => ({
  type: login,
  payload: {
    uid,
    displayName
  }
})
