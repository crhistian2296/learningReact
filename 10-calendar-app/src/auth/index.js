export * from './pages/Login.jsx';
