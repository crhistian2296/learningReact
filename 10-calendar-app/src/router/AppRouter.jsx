import { Navigate, Route, Routes } from 'react-router-dom';
import { LoginPage } from '../auth';
import { CalendarPage } from '../calendar/pages/CalendarPage';

const AppRouter = () => {
  const authStatus = 'authenticated'; // 'authenticated' | 'not-authenticated'
  return (
    <>
      <Routes>
        {authStatus === 'not-authenticated' ? (
          <Route path='/auth/*' element={<LoginPage />}></Route>
        ) : (
          <Route path='/*' element={<CalendarPage />}></Route>
        )}
        <Route path='/*' element={<Navigate to={'/auth/login'} />}></Route>
      </Routes>
    </>
  );
};

export default AppRouter;
