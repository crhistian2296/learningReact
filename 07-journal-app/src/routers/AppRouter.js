import { useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { BrowserRouter as Router, Switch } from 'react-router-dom';
import { login } from '../actions/auth';
import { startLoadingNotes } from '../actions/notes';
import JournalScreen from '../components/journal/JournalScreen';
import { firebase } from '../firebase/firebase-config';
import AuthRouter from './AuthRouter';
import { PrivateRoutes } from './PrivateRoutes';
import { PublicRoutes } from './PublicRoutes';

const AppRouter = () => {
  const [checking, setChecking] = useState(true);
  const [loggedIn, setLoggedIn] = useState(false);
  const dispatch = useDispatch();

  useEffect(() => {
    firebase.auth().onAuthStateChanged(async (user) => {
      if (user?.uid) {
        dispatch(login(user.uid, user.displayName));
        setLoggedIn(true);

        dispatch(startLoadingNotes(user.uid));
      } else setLoggedIn(false);
      setChecking(false);
    });
  }, [dispatch, setChecking]);

  if (checking) return <h1>Wait...</h1>;

  return (
    <Router>
      <div>
        <Switch>
          <PublicRoutes path='/auth' component={AuthRouter} isAuthenticated={loggedIn} />
          <PrivateRoutes exact path='/' component={JournalScreen} isAuthenticated={loggedIn} />
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
