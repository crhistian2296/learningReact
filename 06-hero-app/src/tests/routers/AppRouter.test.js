import { mount } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import AppRouter from '../../routers/AppRouter';

describe('Pruebas sobre <AppRouter/>', () => {
  test('Debe de mostrar <LoginScreen/> si el usuario no esta logueado', () => {
    const contextValue = { user: { logged: false } };
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text().trim()).toBe('LoginScreen');
  });

  test('Debe de mostrar <MarvelScreen/> si el usuario esta logueado', () => {
    const contextValue = { user: { logged: true, name: 'Crhistian' } };
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text().trim()).toBe('Marvel Screen');
    expect(wrapper.find('.text-info').text().trim()).toBe('Crhistian');
  });
});
