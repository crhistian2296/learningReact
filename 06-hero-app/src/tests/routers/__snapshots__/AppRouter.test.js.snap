// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <AppRouter/> Debe de mostrar <LoginScreen/> si el usuario no esta logueado 1`] = `
<div
  className="main"
>
  <div
    className="m-5"
  >
    <h1>
      LoginScreen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;

exports[`Pruebas sobre <AppRouter/> Debe de mostrar <MarvelScreen/> si el usuario esta logueado 1`] = `
<div
  className="main"
>
  Array [
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand"
          href="/"
          onClick={[Function]}
        >
          Asociaciones
        </a>
        <div
          className="navbar-collapse justify-content-sm-between "
        >
          <div
            className="navbar-nav"
          >
            <a
              className="nav-item nav-link"
              href="/marvel"
              onClick={[Function]}
            >
              Marvel
            </a>
            <a
              className="nav-item nav-link"
              href="/dc"
              onClick={[Function]}
            >
              DC
            </a>
            <a
              className="nav-item nav-link"
              href="/search"
              onClick={[Function]}
            >
              Search
            </a>
          </div>
          <div
            className="navbar-nav"
          >
            <span
              className="nav-item nav-link text-info"
            >
              Crhistian
            </span>
            <button
              className="nav-item nav-link btn logoutLink"
              onClick={[Function]}
            >
              Logout
            </button>
          </div>
        </div>
      </div>
    </nav>,
    <div
      className="container my-3"
    >
      <div>
        <h1>
          Marvel Screen
        </h1>
        <div
          className="row row-cols-xl-3 row-cols-md-2 row-cols-1 g-3 animate__animated animate__fadeIn"
        >
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Spider Man"
                    className="card-img-top"
                    src="assets/heroes/marvel-spider.jpg"
                  />
                </div>
                <div
                  className="col-8 card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Spider Man
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Peter Parker
                     
                  </p>
                  <p
                    className="text-muted"
                  >
                     
                    Amazing Fantasy #15
                     
                  </p>
                  <a
                    href="/hero/marvel-spider"
                    onClick={[Function]}
                  >
                    More..
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Captain America"
                    className="card-img-top"
                    src="assets/heroes/marvel-captain.jpg"
                  />
                </div>
                <div
                  className="col-8 card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Captain America
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Steve Rogers
                     
                  </p>
                  <p
                    className="text-muted"
                  >
                     
                    Captain America Comics #1
                     
                  </p>
                  <a
                    href="/hero/marvel-captain"
                    onClick={[Function]}
                  >
                    More..
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Iron Man"
                    className="card-img-top"
                    src="assets/heroes/marvel-iron.jpg"
                  />
                </div>
                <div
                  className="col-8 card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Iron Man
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Tony Stark
                     
                  </p>
                  <p
                    className="text-muted"
                  >
                     
                    Tales of Suspense #39
                     
                  </p>
                  <a
                    href="/hero/marvel-iron"
                    onClick={[Function]}
                  >
                    More..
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Thor"
                    className="card-img-top"
                    src="assets/heroes/marvel-thor.jpg"
                  />
                </div>
                <div
                  className="col-8 card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Thor
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Thor Odinson
                     
                  </p>
                  <p
                    className="text-muted"
                  >
                     
                    Journey into Myster #83
                     
                  </p>
                  <a
                    href="/hero/marvel-thor"
                    onClick={[Function]}
                  >
                    More..
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Hulk"
                    className="card-img-top"
                    src="assets/heroes/marvel-hulk.jpg"
                  />
                </div>
                <div
                  className="col-8 card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hulk
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Bruce Banner
                     
                  </p>
                  <p
                    className="text-muted"
                  >
                     
                    The Incredible Hulk #1
                     
                  </p>
                  <a
                    href="/hero/marvel-hulk"
                    onClick={[Function]}
                  >
                    More..
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Wolverine"
                    className="card-img-top"
                    src="assets/heroes/marvel-wolverine.jpg"
                  />
                </div>
                <div
                  className="col-8 card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Wolverine
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    James Howlett
                     
                  </p>
                  <p
                    className="text-muted"
                  >
                     
                    The Incredible Hulk #180
                     
                  </p>
                  <a
                    href="/hero/marvel-wolverine"
                    onClick={[Function]}
                  >
                    More..
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Daredevil"
                    className="card-img-top"
                    src="assets/heroes/marvel-daredevil.jpg"
                  />
                </div>
                <div
                  className="col-8 card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Daredevil
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Matthew Michael Murdock
                     
                  </p>
                  <p
                    className="text-muted"
                  >
                     
                    Daredevil #1
                     
                  </p>
                  <a
                    href="/hero/marvel-daredevil"
                    onClick={[Function]}
                  >
                    More..
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Hawkeye"
                    className="card-img-top"
                    src="assets/heroes/marvel-hawkeye.jpg"
                  />
                </div>
                <div
                  className="col-8 card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Hawkeye
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Clinton Francis Barton
                     
                  </p>
                  <p
                    className="text-muted"
                  >
                     
                    Tales of Suspense #57
                     
                  </p>
                  <a
                    href="/hero/marvel-hawkeye"
                    onClick={[Function]}
                  >
                    More..
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Cyclops"
                    className="card-img-top"
                    src="assets/heroes/marvel-cyclops.jpg"
                  />
                </div>
                <div
                  className="col-8 card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Cyclops
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Scott Summers
                     
                  </p>
                  <p
                    className="text-muted"
                  >
                     
                    X-Men #1
                     
                  </p>
                  <a
                    href="/hero/marvel-cyclops"
                    onClick={[Function]}
                  >
                    More..
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            className="col animate__animated animate__fadeIn"
          >
            <div
              className="card"
            >
              <div
                className="row g-0"
              >
                <div
                  className="col-4"
                >
                  <img
                    alt="Silver Surfer"
                    className="card-img-top"
                    src="assets/heroes/marvel-silver.jpg"
                  />
                </div>
                <div
                  className="col-8 card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Silver Surfer
                  </h5>
                  <p
                    className="card-text"
                  >
                     
                    Norrin Radd
                     
                  </p>
                  <p
                    className="text-muted"
                  >
                     
                    The Fantastic Four #48
                     
                  </p>
                  <a
                    href="/hero/marvel-silver"
                    onClick={[Function]}
                  >
                    More..
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>,
  ]
</div>
`;
